### ISSUE TITLE
Lerobot special prize

### ISSUE THREAD

**Issue by swampbear** [frdedynamics/project-world-domination-on-weekends#7]
In the [gemma3n compoetition](https://www.kaggle.com/competitions/google-gemma-3n-hackathon/overview) there is a special prize for best utilizing the [lerobot framework](https://github.com/huggingface/lerobot). 

> The LeRobot Prize
> For the most innovative project that turns Gemma 3n into an action model for robotics or automation using the LeRobot framework.

This is the introduction from lerobots github page:
🤗 LeRobot aims to provide models, datasets, and tools for real-world robotics in PyTorch. The goal is to lower the barrier to entry to robotics so that everyone can contribute and benefit from sharing datasets and pretrained models.

🤗 LeRobot contains state-of-the-art approaches that have been shown to transfer to the real-world with a focus on imitation learning and reinforcement learning.

🤗 LeRobot already provides a set of pretrained models, datasets with human collected demonstrations, and simulation environments to get started without assembling a robot. In the coming weeks, the plan is to add more and more support for real-world robotics on the most affordable and capable robots out there.

Any ways. We tried to come up with some ideas, butthe domain knowledge to know what is needed and what is possible. Maybe you have some ideas @MOJOliciousFTW ?

**Comment by MOJOliciousFTW** [comment]
Thinking robot "skills" of some sort and make them tools for gemma. Skills of the fancy sort, and/or simpler services and direct commands.

We have an old yt video showing an assembly task that's easy to understand and verify (gears spin if assembled correct). Maybe use that as task. Can gemma assemble the pieces given a prompt and tools?  https://youtu.be/BnPc4gUXNsw?si=6Ij6Sx90ZCRs8uDp

Can vision be added? Or must it be just robot, no vision?

Sidenote: I've been playing with the thought of putting together one of those huggingface lerobot. Looks very interesting.

**Comment by swampbear** [comment]
Gemma is multimodal and has the following capabilities:
- Audio input: Process sound data for speech recognition, translation, and audio data analysis. [Learn more](https://ai.google.dev/gemma/docs/core/huggingface_inference#audio)
- Visual and text input: Multimodal capabilities let you handle vision, sound, and text to help you understand and analyze the world around you. [Learn more](https://ai.google.dev/gemma/docs/core/huggingface_inference#vision)
- Vision encoder: High-performance MobileNet-V5 encoder substantially improves speed and accuracy of processing visual data. [Learn more](https://developers.googleblog.com/en/introducing-gemma-3n-developer-guide/#mobilenet-v5:-new-state-of-the-art-vision-encoder)
- PLE caching: Per-Layer Embedding (PLE) parameters contained in these models can be cached to fast, local storage to reduce model memory run costs. [Learn more](https://ai.google.dev/gemma/docs/gemma-3n#ple-caching)
- MatFormer architecture: Matryoshka Transformer architecture allows for selective activation of the models parameters per request to reduce compute cost and response times. [Learn more](https://ai.google.dev/gemma/docs/gemma-3n#matformer)
- Conditional parameter loading: Bypass loading of vision and audio parameters in the model to reduce the total number of loaded parameters and save memory resources. [Learn more](https://ai.google.dev/gemma/docs/gemma-3n#conditional-parameter)
- Wide language support: Wide linguistic capabilities, trained in over 140 languages.
32K token context: Substantial input context for analyzing data and handling processing tasks.

Fantasy is the limit here

**Comment by swampbear** [comment]
The more of the capabilites of 3n we use the better imo. With some reason of course
